src/App.js

```diff
@@ -1,37 +1,32 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import * as appActions from './redux/actions';
 import logo from './logo.svg';
 import './App.css';
 import Button from './button';
 import Display from './display';
 import { BUTTON_PROPERTIES_ARRAY } from './constants';
-import { updateResult } from './pressButtonLogic';
 import './calculatorStyle.css';
+import { Provider } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
 
 class App extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {
-            result: ['0'],
-            operators: {
-                array: [],
-                redundant: 0,
-            },
-            default: true,
-            point: false,
-        };
-        this.updateState = this.updateState.bind(this);
+        this.handleUpdateState = this.handleUpdateState.bind(this);
```

Comment:
Please avoid using `bind`. It creates a new function every time it gets called and decreases the performance.